speculos/api/events.py

```diff
@@ -27,6 +33,14 @@ def remove_client(self, client):
 
     def broadcast(self, event):
         self.logger.debug(f"events: broadcasting {event} to ({len(self.clients)}) client(s)")
+        if self.screen_content:
+            # Reset screen content if new event is not below
+            # the last text line of current screen.
+            y_trigger = self.screen_content[-1]["y"] + MIN_LINES_HEIGHT_DISTANCE
+            if event["y"] <= y_trigger:
+                # Reset screen content
+                self.screen_content = []
+        self.screen_content.append(event)
```

Comment:
I don't know what these `event`s are generally speaking, as they're casted as dict which could take whatever form they want, so I'm not comfortable with this code.

If events are always in the form of `{"x": X, "y": Y, ...}`, this code is fine (even if malformed, as I'm not too much into defensive coding I'm cool with a exception poping from here). However this does not seem guaranteed. Is it?

Current attribute naming does not seem to support this: `self.event` is a very generic name, but I don't know. So either event are only `(X, Y)` coordinates and in this case the code should be re-written to highlight this (like `self.events` => `self.screen_events` and `self.screen_content` => `self.current_screen_content`), or the new code may break with legitimate event (`KeyError: 'y'`), and so should manage the cases where `event` is something other than `(X, Y)`.

This would be way clearer with proper typing.

Reply:
I made sure all the events are properly typed as `TextEvent` when generated.